### ISSUE TITLE
Not unique Package.ID generation.

### ISSUE THREAD

**Issue by KotM** [xmppo/Jabber-Net#101]
It seems like both `IQ `and `Message `classes (inherited from `Package`) incorrectly generates ID values when interacting with multiple threads. Sometimes these classes may produce the ID was generated before. 

Tested on latest Jabber-net version too.

Please find below simply UnitTest with reproduction this issue almost constantly. 

```
        [TestMethod]
        public void MultipleIQCreation()
        {
            ConcurrentBag<string> ids = new ConcurrentBag<string>();

            var doc = new XmlDocument();

            //for (int i = 0; i < 500; i++)
            //{
            //    ids.Add((new IQ(doc)).ID);
            //}

            Parallel.For(0, 500, i => ids.Add((new IQ(doc)).ID));

            Assert.AreEqual(500, ids.Count);
            Assert.AreEqual(500, ids.Distinct().Count());
        }
```

**Comment by KotM** [comment]
Workaround it simply -- I've just assigned my own id to the packet. But it would be great to fix it on Jabber-net side. Suppose, the fix would be easy enough, something like `Interlocked.Increment(ref _jabberPacketId)`.

**Comment by ForNeVeR** [comment]
Ah my gosh, this library is full of threading problems. Thanks for your wonderful unit test, I'll take a look.

**Comment by pomoika** [comment]
The bug is in JabberNet/jabber/protocol/Element.cs, method NextID. It should be changed as follows:
```
public static string NextID()
{
    return "JN_" + System.Threading.Interlocked.Increment(ref s_counter).ToString();
}
```

**Comment by ForNeVeR** [comment]
Thanks for your help, it definitely was an issue highlighted by @pomoika. It's fixed now and I've published a new preview release [3.0.0-preview3](https://www.nuget.org/packages/jabber-net/3.0.0-preview3).
